apis/shared-dimensions/lib/domain/shared-dimensions.ts

```diff
@@ -17,12 +18,31 @@ export function getSharedDimensions() {
     `
 }
 
-export function getSharedTerms(sharedDimension: Term) {
+export function getSharedTerms(sharedDimension: Term, freetextQuery: string | undefined, limit = 100) {
+  const term = $rdf.variable('term')
+  const name = $rdf.variable('name')
+
+  let select = SELECT.DISTINCT`${term}`
+    .WHERE`
+      ${sharedDimension} a ${meta.SharedDimension} .
+      ${term} ${schema.inDefinedTermSet} ${sharedDimension} .
+      ${term} ${schema.name} ${name} .
+    `
+
+  if (freetextQuery) {
+    select = select.WHERE`FILTER (
+      REGEX(${name}, "^${freetextQuery}", "i")
+    )`
```

Comment:
I quickly realised that this may give unexpected results with multilingual labels.

Should we take `Accept-Language` request header and filter by the requested languages, possibly with a fallback on a preconfigured lang (`de`/`en`)?

Reply:
We don't currently have a multi-lingual app, would you use the language selected for the cube preview?

Reply:
Can we simply stick to the label currently shown in the list? How is it done right now?

Reply:
> How is it done right now?

The form indeed defaults to labels matching browser language (`window.navigator.languages`)

Filtering is not the same. If you filter on wrong language, or all combined, you might get an unexpected result back